exp/addons/api/v1alpha3/clusterresourcesetbinding_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ANCHOR: ResourceBinding
+
+// ResourceBinding keeps status of a resource that is applied to the cluster that ClusterResourceSetBinding belongs to.
+type ResourceBinding struct {
+	// Hash is the hash of a resource's data. This can be used to decide if a resource is changed.
+	// For "ApplyOnce" ClusterResourceSet.spec.strategy, this is no-op as that strategy does not act on change.
+	Hash string `json:"hash,omitempty"`
+	// LastAppliedTime identifies when this resource was last applied to the cluster.
+	// +optional
+	LastAppliedTime *metav1.Time `json:"lastAppliedTime,omitempty"`
+	// Applied is to track if a resource is applied to the cluster or not.
+	Applied bool `json:"applied,omitempty"`
```

Comment:
Is this going to be true only once, how do we track which clusters we've applied this resource to? For example let's say I create a new CRS, which matches 3 clusters with label `test: 1`. If I create a new cluster with label `test: 1`, how does CRS know to apply the resources to this cluster but not to the other clusters?

Reply:
I think I've answered my own question looking at the code down, seems each CRSBinding has an ownerRef to a cluster

Reply:
Yes, each cluster has a CRSBinding and CRS controller checks CRSBinding when it decides to apply a resource to a cluster.